openmetadata-ui/src/main/resources/ui/playwright/e2e/Pages/Teams.spec.ts

```diff
@@ -579,60 +622,143 @@ test.describe('Teams Page', () => {
       )
     ).toHaveClass(/active/);
 
-    const updateTeamResponse = page.waitForResponse('/api/v1/users*');
+    const updateTeamResponse = adminPage.waitForResponse('/api/v1/users*');
 
     // Update the team with the new user
-    await page.locator('[data-testid="selectable-list-update-btn"]').click();
+    await adminPage
+      .locator('[data-testid="selectable-list-update-btn"]')
+      .click();
     await updateTeamResponse;
 
     // Verify the user is added to the team
     await expect(
-      page.locator(`[data-row-key="${userName.toLowerCase()}"]`)
+      adminPage.locator(`[data-row-key="${userName.toLowerCase()}"]`)
     ).toBeVisible();
 
-    await page
+    await adminPage
       .locator(`[data-row-key="${userName.toLowerCase()}"]`)
       .getByTestId('remove-user-btn')
       .click();
 
-    const updatedTeamResponse = page.waitForResponse('api/v1/users*');
+    const updatedTeamResponse = adminPage.waitForResponse('api/v1/users*');
 
-    await page.getByRole('button', { name: 'confirm' }).click();
+    await adminPage.getByRole('button', { name: 'confirm' }).click();
     await updatedTeamResponse;
 
     await expect(
-      page.locator(`[data-row-key="${userName.toLowerCase()}"]`)
+      adminPage.locator(`[data-row-key="${userName.toLowerCase()}"]`)
     ).not.toBeVisible();
 
     await afterAction();
   });
 });
+
+test.describe('Teams Page with EditUser Permission', () => {
+  test.slow(true);
+
+  test.beforeAll('Setup pre-requests', async ({ browser }) => {
+    const { apiContext, afterAction } = await performAdminLogin(browser);
+    await editOnlyUser.create(apiContext);
+
+    const id = uuid();
+    await policy.create(apiContext, EDIT_USER_FOR_TEAM_RULES);
+    await role.create(apiContext, [policy.responseData.name]);
+
+    team = new TeamClass({
+      name: `PW%edit-user-team-${id}`,
+      displayName: `PW Edit User Team ${id}`,
+      description: 'playwright edit user team description',
+      teamType: 'Group',
+      users: [editOnlyUser.responseData.id],
+      defaultRoles: role.responseData.id ? [role.responseData.id] : [],
+    });
+    await team.create(apiContext);
+    await team2.create(apiContext);
+    await user.create(apiContext);
+    await user2.create(apiContext);
+    await afterAction();
+  });
+
+  test.afterAll('Cleanup', async ({ browser }) => {
+    const { apiContext, afterAction } = await performAdminLogin(browser);
+    await user.delete(apiContext);
+    await user2.delete(apiContext);
+    await editOnlyUser.delete(apiContext);
+    await team.delete(apiContext);
+    await team2.delete(apiContext);
+    await policy.delete(apiContext);
+    await role.delete(apiContext);
+    await afterAction();
+  });
+
+  test.beforeEach('Visit Home Page', async ({ editOnlyUserPage }) => {
+    await redirectToHomePage(editOnlyUserPage);
+    const fetchOrganizationResponse = editOnlyUserPage.waitForResponse(
+      '/api/v1/teams?parentTeam=Organization&include=all&fields=**'
```

Comment:
Can we please make sure we are awaiting on the right API here? Since I remember there are more than one call made to fetch team details.